### ISSUE TITLE
Disabling AFP doesn't actually disable it

### ISSUE THREAD

**Issue by jonpwilson** [rockstor/rockstor-core#2123]
As reported by anborn on the [forums](https://forum.rockstor.com/t/unable-to-deactivate-afp-service-in-web-ui/6727), I'm creating the issue as I have the same problem.

Version: 3.9.1-16

Steps to recreate:

1. Login to the GUI
2. Head to 'System' -> 'Services' and toggle 'AFP'
3. Wait a second and it will refresh and turn itself back on.

![rockstor_AFP_disable](https://user-images.githubusercontent.com/6515084/73781583-6b53cd80-4788-11ea-901c-a517a232155a.gif)

Looking in /var/log/messages shows that AFP is being restarted.

![image](https://user-images.githubusercontent.com/6515084/73781636-8a525f80-4788-11ea-9947-23ba5ad59a92.png)

**Comment by jonpwilson** [comment]
Just a bit more info (and hopefully jog someone's memory!) but it appears that something is restarting the service in the background:

Running `systemctl stop netatalk.service && systemctl disable netatalk.service && watch systemctl status netatalk`: 

![rockstor_netatalk_restart](https://user-images.githubusercontent.com/6515084/73820474-c7027300-47e9-11ea-9283-26a7eab6f717.gif)

**Comment by phillxnet** [comment]
@jonpwilson Thanks for starting out this issue, I don't think we have a lot of AFP users and I was thinking of scrapping the entire service as per openmediavault (needs link) but if we can fix this and the service is workable within our openSUSE offerings then it might be nice from a continuity point of view to try and keep it around. However I can't at the moment spend much time on this and given Apple now prioritises SMB over AFP and with newer Samba able to do time machine functionality I am tempted to fix all AFP related issues by simply removing it whole sale. And this would best be done before we release our stable variant within the 'Built on openSUSE' rpm repos.

Linking out our issue re Time Machine support via SMB:

SMB Time Machine support #1910

Linking to other issues that may even be related to this continuous cycling:

[BUGS] Default ulimit setting too low #1656
with duplicate:
Exception: Exception while running command(['/usr/bin/hostnamectl', '--static']): [Errno 24] Too many open files #2020

I can off some initial pointers at least from an associated mechanisms point of view.

> Just a bit more info (and hopefully jog someone's memory!) but it appears that something is restarting the service in the background:

I remember the following addition, it may not be the origin of this bug but it may be an active participant, if unwittingly.

https://github.com/rockstor/rockstor-core/pull/1735/files

If nothing else it may point to some of the mechanisms involved. I'm also fairly sure, can't look properly for now, that this particular type of 'smart' django.db.models.signals event driven updater is a one-off within Rockstor which may help with narrowing down the cause of this issue; given we don't see this behaviour with any of the other services (hopefully).

I'm still tempted to not serve AFP at all though as from memory I think netatalk is no longer well maintained and with SMB now favoured by Apple, then why should we? It would significantly reduce our support requirements and we are a small development team, currently. And we have had very few contributions in that area in comparison to the SMB related service. But then if it only needs a bug fix and we can continue to serve it, but without much contributor interest and with quite a few likely related bugs and there now being alternatives we might be better off dumping all AFP code and docs. In my limited experience, with older OSX, while preparing the AFP docs it had shaky reliability at best. Maybe this is a forum topic but there is still the low number of folks that have actively involved themselves in our AFP service / bug squashing. And I'm strongly in favour of doing a few things very well rather than a lot poorly, or in an unsustainable manner; ie fire fighting across many services.

So in short a complete removal of AFP could 'fix' this. And the timing might be good. Me and @FroggyFlox  have discussed this. Seems a shame but if it's not missed and we get to do better in SMB then everybody might win :).

**Comment by jonpwilson** [comment]
Hi @phillxnet,

Sorry it took so long to reply but the above sounds sensible to me. 

The original issue reported in to forum can be 'fixed' in the short term by deleting the `/etc/netatalk/afp.conf` file and killing netatalk, without the AFP config there it fails to reload and stops trying (at least for me).

I'd had a quick look at the code and wound up at the same file as you mentioned in #1735 - if it is that, it's interesting that it's taken 2 and a half years to notice! I'll have another look through, though if you're considering removing it, I guess it's not worth spending **too** much time on it?

Thanks

**Comment by phillxnet** [comment]
> if it is that, it's interesting that it's taken 2 and a half years to notice!

I know, but I just don't think we have that many Mac users. No way to tell really though, but given the number of folks that have stepped up to fix AFP stuff I'm thinking there just arn't that many. But on the other hand the SMB Time Machine issue has gotten some attention.

> I'll have another look through, though if you're considering removing it, I guess it's not worth spending **too** much time on it?

Agreed, I think it's pretty much destined to be removed now. I've spoken again with @FroggyFlox and that does seem to be the consensus. But it would be great if we could first offer an alternative via an extended capability to our Web-UI configure of our now 'new enough' Samba to serve the previous AFP Time Machine function hence linking to long standing:

SMB Time Machine support #1910

This might just take the form of a tick option or two that sets this up for a specific share. But I just haven't looked into it in that much detail just yet.

As then at least we don't leave folks with no other option. But the samba route for Time Machine is only open to the newer OSX versions but then time is relentless that way. And if the upstream project is no longer maintained adequately then our choices are very narrow.

Thanks again for tending to this issue but I'm definitely thinking we are better off concentrating on improving / bug fixing our far more commonly used samba and NFS subsystems than patching up an un-maintained AFP system.

**Comment by phillxnet** [comment]
@jonpwilson Just a bit of house keeping.
Re:
> I was thinking of scrapping the entire service as per openmediavault (needs link) but if we can fix this and the service is workable within our openSUSE offerings

and

> Agreed, I think it's pretty much destined to be removed now. I've spoken again with @FroggyFlox and that does seem to be the consensus.
> But it would be great if we could first offer an alternative via an extended capability to our Web-UI configure of our now 'new enough' Samba to serve the previous AFP Time Machine function hence linking to long standing:
> 
> SMB Time Machine support #1910

#1910 has now been resolved by @FroggyFlox  via:

Enable Time Machine support via Samba. Fixes #1910. #2144

and we have also now taken the anticipated deprecation/removal of netatalk, again by @FroggyFlox  in:

Deprecate Netatalk/AFP. Fixes #2146. #2148

So I'm closing this issue as now redundant. Thanks for opening this in the first place as it has served well as a focus for our anticipated deprecation of AFP due to upstream support issues and the comparative lack of engagement we received for bugs in this area of Rockstor.

Thanks to all who helped in clarifying this issue.
